src/core/thread/mle_router.cpp

```diff
@@ -1422,6 +1431,13 @@ Error MleRouter::HandleAdvertisement(RxInfo &aRxInfo)
         ExitNow();
 
     case kRoleRouter:
+        if (mMulticastLinkRequestDelay > 0 && route.IsRouterIdSet(mRouterId))
+        {
+            mMulticastLinkRequestDelay = 0;
+            IgnoreError(SendLinkRequest(nullptr));
+            ExitNow(error = kErrorNoRoute);
```

Comment:
I wanted to avoid sending a unicast link request in the code below. I changed the code to remove this `ExitNow()` and added a check to suppress sending a Link Request is `mChallengeTimeout` non-zero.